Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -0,0 +1,645 @@
+      SourceLink sl = *it;
+      auto geoId = sl.geometryId();
+      inputMeasurements.emplace(geoId, std::move(sl));
+    }
+    std::cout << "inputMeasurements.size() = " << inputMeasurements.size()
+              << std::endl;
+    //    for (const auto &p : inputMeasurements) {
+    //      std::cout << "inputMeasurements.map-loop = " <<
+    //      p.second.geometryId() << std::endl;
+    //    }
+
+    /// Fully understand Aborter, Actor, Result later
+    // Create the ActionList and AbortList
+    using GX2FAborter = Aborter<parameters_t>;
+    using GX2FActor = Actor<parameters_t>;
+
+    using GX2FResult = typename GX2FActor::result_type;
+    using Actors = Acts::ActionList<GX2FActor>;
+    using Aborters = Acts::AbortList<GX2FAborter>;
+
+    using PropagatorOptions = Acts::PropagatorOptions<Actors, Aborters>;
+
+    const size_t reducedMatrixSize = 4;
+    Acts::CurvilinearTrackParameters params = sParameters;
+    BoundVector deltaParams = BoundVector::Zero();
+    double chi2sum = 0;
+    BoundMatrix aMatrix = BoundMatrix::Zero();
+    BoundVector bVector = BoundVector::Zero();
+
+    std::cout << "params:\n" << params << std::endl;
+
+    /// Actual Fitting /////////////////////////////////////////////////////////
+    std::cout << "\nStart to iterate" << std::endl;
+
+    // Iterate the fit and improve result. Abort after n steps or after
+    // convergence
+    for (size_t nUpdate = 0; nUpdate < gx2fOptions.nUpdateMax; nUpdate++) {
+      std::cout << "\nnUpdate = " << nUpdate + 1 << "/"
+                << gx2fOptions.nUpdateMax << "\n"
+                << std::endl;
+
+      // update params
+      params.parameters() += deltaParams;
+      std::cout << "updated params:\n" << params << std::endl;
+
+      // set up propagator and co
+      Acts::GeometryContext geoCtx = gx2fOptions.geoContext;
+      Acts::MagneticFieldContext magCtx = gx2fOptions.magFieldContext;
+      // Set options for propagator
+      PropagatorOptions propagatorOptions(geoCtx, magCtx);
+      auto& gx2fActor = propagatorOptions.actionList.template get<GX2FActor>();
+      gx2fActor.inputMeasurements = &inputMeasurements;
+      gx2fActor.extensions = gx2fOptions.extensions;
+
+      typename propagator_t::template action_list_t_result_t<
+          CurvilinearTrackParameters, Actors>
+          inputResult;
+
+      auto& r = inputResult.template get<Gx2FitterResult<traj_t>>();
+
+      r.fittedStates = &trackContainer.trackStateContainer();
+      // propagate with params and return jacobians and residuals
+      auto result = m_propagator.template propagate(params, propagatorOptions,
+                                                    std::move(inputResult));
+
+      // TODO Improve Propagator + Actor [allocate before loop], rewrite
+      // makeMeasurements
+      auto& propRes = *result;
+      auto gx2fResult = std::move(propRes.template get<GX2FResult>());
+
+      //      std::cout << "gx2fResult.collectorResiduals.size() = "
```

Comment:
generally I would say to make this `ACTS_VERBOSE`, e.g. with the following pattern (see https://en.cppreference.com/w/cpp/io/basic_stringstream):

```c++
std::stringstream s;
for (auto x : y) {
  s << x;
}
s << std::endl;
ACTS_VERBOSE("abc" << s.str());
```

Then you could enable the printing with the log-level, and you don't need to recompile to print this.